shim/opentelemetry-opentracing-shim/src/opentelemetry/shim/opentracing_shim/__init__.py

```diff
@@ -670,13 +673,17 @@ def start_span(
         if start_time_ns is not None:
             start_time_ns = util.time_seconds_to_ns(start_time)
 
-        span = self._otel_tracer.start_span(
-            operation_name,
-            context=parent_span_context,
-            links=valid_links,
-            attributes=tags,
-            start_time=start_time_ns,
-        )
+        kwargs = {
+            "context": parent_span_context,
+            "links": valid_links,
+            "attributes": tags,
+            "start_time": start_time_ns,
+        }
+        if util.opentracing_to_opentelemetry_kind_tag(tags) is not None:
+            kwargs["kind"] = util.opentracing_to_opentelemetry_kind_tag(tags)
+            del tags[SPAN_KIND]
```

Comment:
What is the purpose of `del` here?

Reply:
The idea is : If we set it both in tag and in kind it is a redundancy of informations . This indeed happens when using jaeger thrift exporter where I see a span with span.kind tag duplicated 

## Small analysis of exporters : 
In jaeger exporter (exporter I use), the kind is added to tag so I was planning to remove
* exporter-jaeger-thrift : https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-jaeger-thrift/src/opentelemetry/exporter/jaeger/thrift/translate/__init__.py#L218-L220
* exporter-jaeger-grpc : https://github.com/open-telemetry/opentelemetry-python/blob/main/exporter/opentelemetry-exporter-jaeger-proto-grpc/src/opentelemetry/exporter/jaeger/proto/grpc/translate/__init__.py#L296-L300

In opencensus,otlp and zipkin, it seems to be specific field so should not be duplication issues

Reply:
I agree with the not duplication of data, but perhaps it would be better suited for the exporters that deal with the data rather than the shim itself. The shim should not modify any of the data coming in.

Reply:
Agreed . I removed the deletion from tracer and will probably open PR on jaeger exporter